### PR TITLE
Move code generation logic from main() to gen/

### PR DESCRIPTION
**Pull request by abhinav**
This allows golden_test to do code generation in the same process.

### CODE REVIEW

**Comment by abhinav** [comment]
Failure:

```
goveralls -service=travis-ci -v ./ast/... ./compile/... ./gen/... ./idl/... ./protocol/... ./wire/... .
exit status 1: cannot use test profile flag with multiple packages
```

File: gen/generate.go [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+func constantNames(constants map[string]*compile.Constant) []string {
+	names := make([]string, 0, len(constants))
+	for name := range constants {
+		names = append(names, name)
+	}
+	sort.Strings(names)
+	return names
+}
+
+// constantNames sorts the keys of a map of types in a deterministic order.
```
Comment (prashantv) [inline gen/generate.go]: stale function name in comment

**Comment by prashantv** [comment]
lgtm, not sure what's up with goveralls, wonder if they broke something recently..

**Comment by abhinav** [comment]
updated

**Comment by prashantv** [comment]
Yay for working coveralls, lgtm.

**Pull request merged**